[PATCH] fix linker variable

The linker variable LD was always set to 'ld' instead of the value from
the environment. This breaks any linking, especially when dealing with
a cross-compiler toolchain.
The fix takes the CC variable and changes the 'gcc' extension to 'ld'.

Signed-off-by: Roeland Van Praet <roelvp@mind.be>
---
diff -ruN a/autoconf/make/vars.mk b/autoconf/make/vars.mk
--- a/autoconf/make/vars.mk	2010-03-11 00:43:19.000000000 +0100
+++ b/autoconf/make/vars.mk	2011-10-06 15:47:18.000000000 +0200
@@ -30,7 +30,7 @@
 @SET_MAKE@
 SHELL = /bin/sh
 CC = @CC@
-LD = ld
+LD = $(CC:gcc=ld)
 DO_GZIP = @DO_GZIP@
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
